server/src/com/thoughtworks/go/server/security/ArtifactsFilter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.security;
+
+import org.springframework.security.AccessDeniedException;
+import org.springframework.security.ui.SpringSecurityFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+import static org.springframework.security.ui.FilterChainOrder.EXCEPTION_TRANSLATION_FILTER;
+
+public class ArtifactsFilter extends SpringSecurityFilter {
+
+    @Override
+    protected void doFilterHttp(HttpServletRequest request, HttpServletResponse response, FilterChain chain) throws IOException, ServletException {
+
+        if (isRequestFromArtifact(request) && !requestingAnArtifact(request)) {
```

Comment:
@4cad  --- Thanks for the feedback. This filter will block request that came from an artifact and requesting non artifact resource. We tried following scenarios:

1) Accessing login or home page from an artifact: Filter will block any non artifact calls from an artifact as I mentioned above.
     * In following screenshot `attack_alert.html` is an artifact accessing login and home page through ajax calls.

        <img width="717" alt="screen shot 2017-06-23 at 11 04 43 am" src="https://user-images.githubusercontent.com/7871209/27467930-17ae243a-5804-11e7-8d15-36f258080aae.png">

        (Assuming, this will anser you question)     


2) Accessing an artifact after session termination:
    a. Open an artifact
    b. From browser use different tab and, end the current session
    c. Reload artifact page[a]

    On reload of artifact page user will be redirected to login page. After successful authentication user will be redirected back to the artifact page
   
3) Accessing artifact from API with credentials: Working as expected

Let me know, If you need more clarification.